File: Demo/Sources/ViewControllers/PhoneField/PhoneFieldViewController.m
//
//  Globus
//

#import "PhoneFieldViewController.h"

@interface PhoneFieldViewController ()

@property(nonatomic, weak) IBOutlet GLBPhoneField* phoneField;
@property(nonatomic, weak) IBOutlet UILabel* regionLabel;
@property(nonatomic, weak) IBOutlet UILabel* phoneNumberLabel;
@property(nonatomic, weak) IBOutlet UILabel* phoneNumberWithoutPrefixLabel;

@end

@implementation PhoneFieldViewController

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    GLBTextStyle* textStyle = [GLBTextStyle new];
    textStyle.font = [UIFont boldSystemFontOfSize:16.0f];
    textStyle.color = UIColor.darkGrayColor;
    _phoneField.textStyle = textStyle;
    
    _phoneField.prefix = @"+";
    [_phoneField setDefaultOutputPattern:@"############"];
    [_phoneField addOutputPattern:@"# (###) ###-##-##" forRegExp:@"^7\\d{10}$" userInfo:@"RU"];
    [_phoneField addOutputPattern:@"## (####) ##-##-##" forRegExp:@"^44\\d{10}$" userInfo:@"UK"];
    
    [self changedPhone:nil];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    [_phoneField becomeFirstResponder];
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)changedPhone:(id)sender {
    NSString* region = _phoneField.userInfo;
    if(region.glb_isString == YES) {
        _regionLabel.text = region;
    } else {
        _regionLabel.text = @"NONE";
    }
    _phoneNumberLabel.text = _phoneField.phoneNumber;
    _phoneNumberWithoutPrefixLabel.text = _phoneField.phoneNumberWithoutPrefix;
}

@end

File: Demo/Sources/ViewControllers/PhoneField/PhoneFieldViewController.swift
//
//  Globus
//

import Globus

class PhoneFieldViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var phoneField: GLBPhoneField!
    @IBOutlet fileprivate weak var regionLabel: UILabel!
    @IBOutlet fileprivate weak var phoneNumberLabel: UILabel!
    @IBOutlet fileprivate weak var phoneNumberWithoutPrefixLabel: UILabel!
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let textStyle = GLBTextStyle.init()
        textStyle.font = UIFont.boldSystemFont(ofSize: 16)
        textStyle.color = UIColor.darkGray
        self.phoneField.textStyle = textStyle
        
        self.phoneField.prefix = "+"
        self.phoneField.setDefaultOutputPattern("############")
        self.phoneField.addOutputPattern("# (###) ###-##-##", forRegExp: "^7\\d{10}$", userInfo: "RU")
        self.phoneField.addOutputPattern("## (####) ##-##-##", forRegExp: "^44\\d{10}$", userInfo: "UK")
        
        self.changedPhone(self.phoneField)
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        self.phoneField.becomeFirstResponder()
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func changedPhone(_ sender: Any) {
        if let region = self.phoneField.userInfo as? String {
            self.regionLabel.text = region
        } else {
            self.regionLabel.text = "NONE"
        }
        self.phoneNumberLabel.text = self.phoneField.phoneNumber()
        self.phoneNumberWithoutPrefixLabel.text = self.phoneField.phoneNumberWithoutPrefix()
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "PhoneFieldViewController-Swift"
    }
    
}
